Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1072,6 +1473,608 @@ def get_cim_mapping_field_command():
+        failed_notables (list): The list of notables that failed to enrich
+
+    """
+    enriched_notables, not_yet_enriched_notables, failed_notables = [], [], []
+
+    for notable in notables:
+        if notable.status == NotableStatus.ENRICHED:
+            enriched_notables.append(notable)
+        elif notable.status == NotableStatus.NOT_YET_ENRICHED:
+            not_yet_enriched_notables.append(notable)
+        elif notable.status == NotableStatus.CREATED_INCIDENT_ENRICH_FAILURE:
+            failed_notables.append(notable)
+
+    return enriched_notables, not_yet_enriched_notables, failed_notables
+
+
+def enrich_fetched_notable(service, notable, num_enrichment_events, incidents, cache_object):
+    """ Enriches fetched notable. Three enrichments possible: Drilldown, Asset & Identity.
+    If all enrichment type executions were unsuccessful, creates a regular incident, Otherwise updates the
+    integration context for the next fetch to handle the open enrichment.
+
+    Args:
+        service (splunklib.client.Service): Splunk service object
+        Possible values are: Drilldown, Asset & Identity.
+        incidents (list): The incident to be submitted at the end of the run.
+        num_enrichment_events (int): The maximal number of events to return per enrichment type.
+        notable (Notable): The notable.
+        cache_object (Cache): The enrichment mechanism cache object
+
+    """
+    notable.data.update({enrichment_type: [] for enrichment_type in ENABLED_ENRICHMENTS})
+    enrichment = Enrichment(notable)
+
+    if DRILLDOWN_ENRICHMENT in ENABLED_ENRICHMENTS:
+        splunk_job = drilldown_enrichment(service, enrichment.notable.data, num_enrichment_events)
+        maintain_enrichment_metadata(enrichment, splunk_job, DRILLDOWN_ENRICHMENT)
+    if ASSET_ENRICHMENT in ENABLED_ENRICHMENTS:
+        splunk_job = asset_enrichment(service, enrichment.notable.data, num_enrichment_events)
+        maintain_enrichment_metadata(enrichment, splunk_job, ASSET_ENRICHMENT)
+    if IDENTITY_ENRICHMENT in ENABLED_ENRICHMENTS:
+        splunk_job = identity_enrichment(service, enrichment.notable.data, num_enrichment_events)
+        maintain_enrichment_metadata(enrichment, splunk_job, IDENTITY_ENRICHMENT)
+
+    if any(job.status == JobStatus.IN_PROGRESS for job in enrichment.jobs):
+        enrichment.status = EnrichmentStatus.DONE_ENRICHING
+        notable.status = NotableStatus.ENRICHED
+        cache_object.enrichments.append(enrichment)
+        demisto.info('Submitted enrichment request to Splunk for enrichment {}'.format(enrichment.id))
+    else:
+        enrichment.status = EnrichmentStatus.CREATED_INCIDENT_ENRICH_FAILURE
+        notable.status = NotableStatus.CREATED_INCIDENT_ENRICH_FAILURE
+        notable.data[SUCCESSFUL_ENRICHMENT] = False
+        incidents.append(notable_to_incident(notable.data))
+
+
+def maintain_enrichment_metadata(enrichment, splunk_job, job_type):
```

Comment:
I think we don't need this function, see my comment on the `enrich_fetched_notable` function.